### PR TITLE
os/bluestore: Prealloc memory avoid realloc in list_collection.

### PR DESCRIPTION
**Pull request by majianpeng**
Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -7121,6 +7121,7 @@ int BlueStore::getattrs(
 
 int BlueStore::list_collections(vector<coll_t>& ls)
 {
+  ls.reserve(coll_map.size());
   RWLock::RLocker l(coll_lock);
```
Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: can you move this down a line under the coll_lock?  It's not safe to call size() outside of the lock.

Comment (majianpeng) [reply src/os/bluestore/BlueStore.cc]: alloc memory may need long time and i don't hope by coll_lock. And if size() increase is small conceptual event. We don't consider this.

**Comment by majianpeng** [comment]
update by your suggestion.

**Comment by tchaikov** [comment]
the failures are either caused by #19117 or tracked by http://tracker.ceph.com/issues/9356 .

**Pull request merged**